### DISCUSSION TITLE
Question about Xarray's minimum versions policy

### DISCUSSION THREAD

**Discussion by maxrjones** [pydata/xarray#7482] [Q&A] [answered]
Hi all, I'm hoping to get some clarification about Xarray's minimum version policy in order to understand the implications for downstream packages (i.e., whether we should follow NEP-29, Xarray’s rolling policy, or something else).

[https://github.com/pydata/xarray/pull/7461](https://github.com/pydata/xarray/pull/7461) removed support for Python 3.8, meaning that all downstream packages will need to drop 3.8 since the PR changed the code to rely on features only in 3.9+. I am confused because the timing seems to be between the [Xarray’s rolling policy](https://docs.xarray.dev/en/stable/getting-started-guide/installing.html#minimum-dependency-versions) (24 months of support for 3.8 would end on Oct 14, 2021) and [NEP-29](https://numpy.org/neps/nep-0029-deprecation_policy.html) (42 months of support would end on April 14, 2023). I think this may relate to the [proposed revision for NEP-29](https://github.com/numpy/numpy/pull/14086#issuecomment-649287648) being [updated in Xarray](https://github.com/pydata/xarray/pull/4907) but not NumPy's docs, though I may be missing some history.

Really appreciate all the hard work the Xarray team puts into the library and user support! Thanks for any guidance you can provide here.

**Comment by keewis** [comment, marked as answer]
I'm sorry nobody ever replied here, but this has come up again and is being discussed in #7765.

As I pointed out there, the rolling release policy actually means "at least 24 months", which for python with its regular release cadence of about 12 months results in a effective rolling release window of 36 months (drop when the next python version has been released more than 24 months ago, so we were able to drop python 3.8 after Oct 5, 2022). I guess we don't make that intention very clear in the policy...
